app/models/grant_submission/response.rb

```diff
@@ -49,19 +51,18 @@ class Response < ApplicationRecord
     validate :validate_by_response_type
     validate :response_if_mandatory, if: -> { question.is_mandatory? && (submission.submitted? || submission&.user_submitted_state == 'submitted') }
     validate :attachment_is_valid,   if: -> { document.attached? }
+    validate :attachment_not_removed_on_submit, if: -> () { question.response_type == 'file_upload' && submission&.user_submitted_state == 'submitted' && remove_document == '1' }
 
-    include DateOptionalTime
-    has_date_optional_time(datetime_comp: :datetime_val, has_time_comp: :boolean_val)
+    after_save :purge_document, if: -> () { question.response_type == 'file_upload' }
```

Comment:
To avoid this being called for all submissions, add `&& remove_document == "1"` to the `if` lambda